src/Aspire.Hosting.Azure.Storage/AzureStorageExtensions.cs

```diff
@@ -326,10 +294,43 @@ public static IResourceBuilder<AzureBlobStorageContainerResource> AddBlobContain
         blobContainerName ??= name;
 
         AzureBlobStorageContainerResource resource = new(name, blobContainerName, builder.Resource);
-
         builder.Resource.Parent.BlobContainers.Add(resource);
 
-        return builder.ApplicationBuilder.AddResource(resource);
+        BlobServiceClient? blobServiceClient = null;
+        builder.ApplicationBuilder.Eventing.Subscribe<ConnectionStringAvailableEvent>(resource, async (@event, ct) =>
+        {
+            var parentResource = resource.Parent;
+            var parentConnectionString = await parentResource.ConnectionStringExpression.GetValueAsync(ct).ConfigureAwait(false);
+            if (parentConnectionString is null)
+            {
+                throw new DistributedApplicationException($"{nameof(ConnectionStringAvailableEvent)} was published for the '{parentResource.Name}' resource but the connection string was null.");
+            }
+
+            blobServiceClient = CreateBlobServiceClient(parentConnectionString);
+        });
+
+        var healthCheckKey = $"{resource.Name}_check";
+        var healthCheckRegistration = new HealthCheckRegistration(
+            healthCheckKey,
+            sp =>
+            {
+                _ = blobServiceClient ?? throw new InvalidOperationException("BlobServiceClient is not initialized.");
+
+                var blobContainerClient = blobServiceClient.GetBlobContainerClient(blobContainerName);
+                blobContainerClient.CreateIfNotExists();
```

Comment:
This is called every time the health check runs. I don't think that's what we want.

Reply:
@RussKie - can you explain why we can't follow the same pattern we are doing for CosmosDB and other databases (Sql Server and Postgres)?